File: backend/app/models/customer.py
from sqlalchemy import Column, Integer, String, Text, ForeignKey, TIMESTAMP
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.core.database import Base

class Customer(Base):
    __tablename__ = "customers"

    customer_id = Column(Integer, primary_key=True, index=True)
    society_id = Column(Integer, ForeignKey("society.society_id"), nullable=False)
    full_name = Column(String(100), nullable=True)
    pan = Column(String(20), unique=True, nullable=True)
    phone = Column(String(20))
    email = Column(String(100))
    address = Column(Text)
    kyc_status = Column(String(20), default="PENDING")
    created_at = Column(TIMESTAMP, server_default=func.now())

    society = relationship("Society", back_populates="customers")
    sales = relationship("Sale", back_populates="customer")

File: frontend/src/types/customer.ts
// types/customer.ts
export type KYCStatus = "PENDING" | "DONE"

export interface Customer {
  customer_id: number
  society_id: number
  full_name: string | null
  pan: string | null
  phone: string | null
  email: string | null
  address: string | null
  kyc_status: KYCStatus
  created_at: string
}

export interface CreateCustomerDTO {
  society_id: number
  full_name: string
  pan: string
  phone?: string | null
  email?: string | null
  address?: string | null
}

export interface UpdateCustomerDTO{
  full_name?: string | null
  phone?: string | null
  email?: string | null
  address?: string | null
  kyc_status?: KYCStatus | null
}

export interface CoApplicantResponse {
  coapplicant_id: number
  customer_id: number
  full_name: string
  pan?: string | null
  phone?: string | null
  email?: string | null
  address?: string | null
  kyc_status: KYCStatus
}

export interface CoApplicantCreate {
  customer_id: number
  full_name: string
  pan?: string | null
  phone?: string | null
  email?: string | null
  address?: string | null
}

export interface CoApplicantUpdate {
  full_name?: string | null
  pan?: string | null
  phone?: string | null
  email?: string | null
  address?: string | null
  kyc_status?: KYCStatus | null
}

